src/backend/core/api/serializers.py

```diff
@@ -26,22 +26,69 @@ def update(self, instance, validated_data):
         return super().update(instance, validated_data)
 
 
-class UserSerializer(serializers.ModelSerializer):
+class UserCompleteSerializer(serializers.ModelSerializer):
     """Serialize users."""
 
     timezone = TimeZoneSerializerField(use_pytz=False, required=True)
+    name = serializers.SerializerMethodField(read_only=True)
+    email = serializers.SerializerMethodField(read_only=True)
 
     class Meta:
         model = models.User
         fields = [
             "id",
+            "name",
             "email",
             "language",
             "timezone",
             "is_device",
             "is_staff",
         ]
-        read_only_fields = ["id", "email", "is_device", "is_staff"]
+        read_only_fields = ["id", "name", "email", "is_device", "is_staff"]
+
+    def get_name(self, obj):
+        """Return main identity's name."""
+        try:
+            return obj.main_identity[0].name
+        except TypeError:
+            return obj.main_identity.name
+        except IndexError:
+            main_identity = obj.identities.filter(is_main=True).first()
+            return obj.main_identity.name if main_identity else None
+        except AttributeError:
+            return None
+
+    def get_email(self, obj):
+        """Return main identity's email."""
+        try:
+            return obj.main_identity[0].email
+        except TypeError:
+            return obj.main_identity.email
+        except IndexError:
+            main_identity = obj.identities.filter(is_main=True).first()
+            return obj.main_identity.email if main_identity else None
+        except AttributeError:
+            return None
```

Comment:
Nit: these logics could be (a bit) refactored.

Reply:
It's a bit clumsy. Feel free to refacto if you see a cooler way to do it ! I don't